roles/zabbix_proxy/tasks/initialize-mysql.yml

```diff
@@ -17,13 +17,24 @@
 
 # NOTE: Upgrading system-packages with pip is generally a bad idea, but
 # these packaged older versions seems to have a problem with mysql 8 and above
+- name: Check if pymysql should be upgraded
+  ansible.builtin.set_fact:
+    __upgrade_debian_pymysql: "{{ (ansible_facts['distribution'] in ['CentOS', 'Debian', 'Ubuntu'] and ansible_facts['distribution_release'] in ['Core', 'buster', 'bullseye', 'bionic', 'focal']) }}"
+    __upgrade_suse_pymysql: "{{ ansible_facts['os_family'] in ['Suse'] }}"
```

Comment:
I kinda see what you're doing here, and it's clever to give names to the checks. Unfortunately they are a bit misleading, and don't really warrant their own task.

The misleading would be calling the variable upgrade_debian_pymysql and checking if distro is 'CentOS'. The good news is I think we've dropped centos7 for this role, you're free to take out 'CentOS' and 'Core' from the comparison. And also buster and bionic.

Having a task to set_fact, when it's just needed for the `Upgrade pymysql` task is to much, so just put these named checks under a `vars:` section in the `Upgrade pymysql` task.

This task is a work-around for specific versions, and I wouldn't want this to be the default for all future releases of Suse, so you should really pin it down to a release, or major version, or something else.

Final note for context, the when condition that was used is more elaborate than it needed to be, mostly for the convenience of the readers/developers of this playbook. It almost reads "this is a bad idea, but upgrade pymysql when debian-bullseye or ubuntu-focal"

Reply:
I agree that the naming isn't optimal, yeah. 
What do you think about creating a variable in the os specific vars file? That would probably be more idiomatic than using the vars section of a task? Something like:

```yaml
_zabbix_server_pymysql_upgrade:
  "15": true
```

That allows us to control the pymysql upgrade for each major versions.

Reply:
I think that approach would hide the details of when this workaround is applied. Suddenly you have to look up information in a second file. (Also you've now created a lookup table, and you'll need to create an entry for all versions in the os_family.)

I think you're good by just setting

```
- name: Upgrade pymysql
  when: _pymysql_upgrade_suse or _pymysql_upgrade_debian
  pip:
    name: ...
  vars:
    _pymysql_upgrade_suse: "{{ ansible_facts['os_family'] == 'Suse' and ansible_facts['distribution_major_release'] == '15' }}"
    _pymysql_upgrade_debian: "{{ ...
```

Or something to that effect. If it was relevant for more versions for os_family, then the vars file would be the way to go.